cmd/spiffe-helper/main.go

```diff
@@ -23,38 +23,47 @@ func main() {
 	flag.Parse()
 	log := logrus.WithField("system", "spiffe-helper")
 
-	if err := startSidecar(*configFile, *daemonModeFlag, log); err != nil {
+	log.Infof("Using configuration file: %q", *configFile)
+	hclConfig, err := config.ParseConfig(*configFile, *daemonModeFlag, daemonModeFlagName)
+	if err != nil {
+		log.WithError(err).Errorf("failed to parse configuration")
+		os.Exit(1)
+	}
+
+	if err := hclConfig.ValidateConfig(log); err != nil {
+		log.WithError(err).Errorf("invalid configuration")
+		os.Exit(1)
+	}
+
+	spiffeSidecar, err := startSidecar(hclConfig, log)
+	if err != nil {
 		log.WithError(err).Errorf("Error starting spiffe-helper")
 		os.Exit(1)
 	}
 
+	if *hclConfig.DaemonMode && hclConfig.HealthCheck.ListenerEnabled {
+		if err := health.StartHealthServer(hclConfig.HealthCheck, log, spiffeSidecar); err != nil {
```

Comment:
Should this be called before startSidecar(). That starts the sidecar daemon which doesn't exit unless there is an error or cancel().

Reply:
I had the sidecar first because the sidecar is what creates/populates the map of file write statuses and has the logic for whether it's healthy or not. Otherwise the health server won't know what content to serve up.